# Portfolio Project: Strategic Credit Allocation for Financial Cooperatives

## A. Project Foundation

**Analytics Domain:** Financial Analytics (Cooperatives & Loan Institutions)

**Optimization Problem:** Optimal Loan Portfolio Mix under Default Risk and Regulatory Constraints.

**Related Literature:** Ogboi, C., Ogunwale, O., Ogunwole, J. O., & Emordi, N. B. (2025). Application of linear programming model in investment portfolio and loan portfolio optimization. International Journal of Economics, Business and Management Research, 9(5), 447–463. Retrieved from: https://www.researchgate.net/publication/392016603_Application_of_Linear_Programming_Model_in_Investment_Portfolio_and_Loan_Portfolio_Optimization?utm_source=chatgpt.com

## B. Business Scenario: Coop-Balance 2026

**Context:** As the Senior Portfolio Analyst for a provincial Credit Cooperative, I have developed a prescriptive model to optimize the cooperative's quarterly credit disbursement strategy while still adhering to strict instititutional risk maximum. 
The task is to manage **$5,000,000** in new loanable funds for the upcoming quarter. The primary challenge is to distribute this capital across three distinct member segments to maximize interest profit. However, the cooperative must operate within strict "Risk contstraints to protect the member's equity and fulfill social mandates for rural development.

The three segments under consideration are:
1. **Low-Risk Micro-Enterprise Loans:** Stable returns with minimal default history.
2. **Medium-Risk Agricultural Loans:** Essential for the cooperative's mission but subject to seasonal volatility.
3. **High-Risk Personal/Emergency Loans:** High-yield returns but prone to higher delinquency rates.


## C. Model Formulation

### **1. Objective Function**
The goal is to **maximize** the Total Expected Annual Interest Income ($Z$). The interest rates for the segments are 7%, 11%, and 16% respectively.

$$Z = 0.07a_1 + 0.11b_2 + 0.16c_3$$

### **2. Decision Variables**
The Linear Programming (LP) model controls the exact dollar amount allocated to each loan segment:
* $x_1$: Capital allocated to **Low-Risk Micro-Enterprise**
* $x_2$: Capital allocated to **Medium-Risk Agriculture**
* $x_3$: Capital allocated to **High-Risk Personal/Emergency**

### **3. Constraints**
To ensure a balanced and sustainable portfolio, the solution must satisfy the following real-world requirements:

1. **Total Capital Budget:** Total allocations must not exceed the available fund.
   $$a_1 + b_2 + c_3 \le 5,000,000$$

2. **Risk (Expected Default):** Based on historical default rates (1.5%, 4.5%, and 9%), the total expected loss must not exceed **$175,000** (3.5% of total funds).
   $$0.015a_1 + 0.045b_2 + 0.090c_3 \le 175,000$$

3. **Minimum (Social Mandate):** At least **25%** of the portfolio must be dedicated to Agricultural loans to support local farmers.
   $$b_2 \ge 1,250,000$$

4. **Maximum (Risk Value):** To prevent over-exposure to high-risk assets, Personal loans are limited to **20%** of the total budget.
   $$c_3 \le 1,000,000$$

5. **No-Negative Values:** To ensure the solution remains physically and financially possible, all decision variables must be greater or equal to zero.
$$a_1,b_2,c_3 \ge 0$$

In [1]:
import pulp

In [3]:
# 1. Optimization Model
# We use LpMaximize because the objective is to maximize interest profit.
model = pulp.LpProblem("Cooperative_Credit_Allocation", pulp.LpMaximize)

In [4]:
# 2. Define Decision Variables (Continuous USD amounts)
a1 = pulp.LpVariable('Micro_Enterprise_Loans', lowBound=0)

# Social Mandate Minimum
b2 = pulp.LpVariable('Agricultural_Loans', lowBound=1250000) 

#Maximum Risk
c3 = pulp.LpVariable('Personal_Loans', lowBound=0, upBound=1000000)

In [5]:
# 3. Objective Function
model += 0.07*a1 + 0.11*b2 + 0.16*c3, "Total_Expected_Income"

In [6]:
# 4. Constraints: Total Capital Budget
model += a1 + b2 + c3 <= 5000000, "TotalBudgetLimit"

In [8]:
# Expected Default Loss Constraint (Max $175,000)
model += 0.015*a1 + 0.045*b2 + 0.090*c3 <= 175000, "Maximum Risk"

In [9]:
#5. Solve
status = model.solve()

In [12]:
# 6. Output

print(f"--- Strategic Allocation Results ({pulp.LpStatus[status]}) ---")
print(f"Optimal Allocation for Micro-Enterprise: ${a1.varValue:,.2f}")
print(f"Optimal Allocation for Agricultural: ${b2.varValue:,.2f}")
print(f"Optimal Allocation for Personal/Emergency: ${c3.varValue:,.2f}")
print(f"--------------------------------------------------")
print(f"Maximized Total Annual Income: ${pulp.value(model.objective):,.2f}")

--- Strategic Allocation Results (Optimal) ---
Optimal Allocation for Micro-Enterprise: $1,666,666.70
Optimal Allocation for Agricultural: $3,333,333.30
Optimal Allocation for Personal/Emergency: $0.00
--------------------------------------------------
Maximized Total Annual Income: $483,333.33


## D. Interpretation of Optimization Results
This optimization provides the Board of Directors with a rigorous, data-driven framework. It demonstrates that under current default rates, the focus should remain on Agricultural and Micro-enterprise sectors to protect member equity."

### **1. Optimal Strategy Summary**

The Linear Programming model successfully identified an optimal allocation that maximizes interest income at **$483,333.33$ annually**.

The strategic mix consists of:

* **Micro-Enterprise Loans ($a_1$):** $1,666,666.70$ *(33.3% of portfolio)*
* **Agricultural Loans ($b_2$):** $3,333,333.30$ *(66.7% of portfolio)*
* **Personal/Emergency Loans ($c_3$):** $0.00$ *(0% of portfolio)*

### **2. Identification of Binding Constraints**

The analysis reveals that the **Maximum Risk** is a **binding constraint**. The total expected default loss is exactly **$175,000$**, which means the model has fully utilized the allowable risk capacity.

Any increase in the allocation of high-risk Personal loans would require a unreasonable reduction in Agricultural or Micro-Enterprise loan to remain within the risk threshold, which ultimately lower total profits.

### **3. Strategic Recommendations for the Board**

* **Capital Efficiency:**
  The model prioritizes the Agricultural sector, which satisfies the cooperative’s social mandate (exceeding the 25% minimum allocation) while it maintains a strong balance between yield and risk.

* **Risk Management:**
  The exclusion of Personal/Emergency loans suggests that, under the current **9% default rate**, this segment is not a viable contributor to income compared to the more stable Micro-Enterprise and Agricultural sectors.

* **Future Growth:**
  To justify entering the high-risk personal loan market, the cooperative would need to either:
  * Increase its risk tolerance threshold, or
  * Implement stricter credit scoring mechanisms to reduce the default rate from 9%.
